### PR TITLE
Fix blindly re-prompt after consultation 

### LINKED ISSUE #2258: [Adaptive] Input does not run its own recognizer after consultation
This is either a port bug or a core issue in how inputs were updated as part of the events DCR. 

The core issue here seems to be that input’s own recognizer is not run after consultation bubbling wraps up and the input always issues a reprompt. 

```C#
var rootDialog = new AdaptiveDialog(nameof(AdaptiveDialog))
{
    Generator = new TemplateEngineLanguageGenerator(),
    Recognizer = new RegexRecognizer() {
        Intents = new Dictionary<string, string>() {
            { "Interruption", "(?i)interrupt" },
            { "Greeting", "(?i)hi" },
            { "Start", "(?i)start" },
            { "noage", "(?i)no" },
            { "why", "(?i)why" },
            { "reset", "(?i)reset" }
        }
    },
    Rules = new List<IRule> ()
    {
        new ConversationUpdateActivityRule()
        {
            Steps = new List<IDialog>()
            {
                new SendActivity("Hello, I'm the demo bot. \\n \\[Suggestions=start | hi | reset\\]")
            }
        },
        new UnknownIntentRule()
        {
            Steps = new List<IDialog>()
            {
                new SendActivity("Hello, I'm the demo bot. \\n \\[Suggestions=start | hi | reset\\]")
            }
        },
        new IntentRule()
        {
            Intent = "reset",
            Steps = new List<IDialog>() {
                new DeleteProperty()
                {
                    Property = "user.name"
                },
                new SendActivity("Sure. I've reset your profile. \\n \\[Suggestions=reset | hi\\]") 
            }
        },
        new IntentRule() {
            Intent = "Start",
            Steps = new List<IDialog>() {
                new TextInput() {
                    Prompt = new ActivityTemplate("What is your name? \\n \\[Suggestions=reset | why | no | hi | interrupt\\]"),
                    Property = "user.name",
                    AllowInterruptions = true
                },
                new SendActivity("I have {user.name} as your name")
            }
        },
        new IntentRule() {
            Intent = "Interruption",
            Steps = new List<IDialog>() {
                // short circuiting Interruption so consultation is terminated. 
                new SendActivity("In Interruption..."),
                // request the active input step to re-process user input. 
                new SetProperty()
                {
                    Property = "turn.StepCount",
                    Value = "-1"
                }
            }
        },
        new IntentRule() {
            Intent = "Greeting",
            Steps = new List<IDialog>() {
                new SendActivity("Hi, I'm the test bot! \\n \\[Suggestions=start | hi | reset\\]")
            }
        },
        new IntentRule()
        {
            Intent = "noage",
            Steps = new List<IDialog>()
            {
                new SendActivity("Sure, no problem. I'll set your name to 'Human'. you can say reset to start over"),
                new SetProperty()
                {
                    Property = "user.name",
                    Value = "'Human'"
                }
            }
        },
        new IntentRule()
        {
            Intent = "why",
            Steps = new List<IDialog>()
            {
                new SendActivity("I need your name to be able to address you correctly")
            }
        }
    }
};
```

**Expected behavior:**

```markdown
> Flow #1
user: start
bot: what is your name?
user: why
bot: I need your name to be able to address you correctly
bot: what is your name?
user: Hi
bot: Hi, I’m the test bot!
user: reset
bot: Sure. I’ve reset your profile.
bot: what is your name?
user: no
bot: Sure, no problem. I’ll set your name to ‘Human’. you can say reset to start over
bot: I have Human as your name

> Flow #2
user: start
bot: what is your name?
user: interrupt
bot: In Interruption…
bot: I have **Interruption** as your name
```

**Current behavior** 

```markdown
user: start
bot: what is your name?
user: no
bot: Sure, no problem, I'll set your name to 'Human'. you can say reset to start over
bot: what is your name?
user: interrupt
bot: In interruption....
bot: I have **Human** as your name
```

### PR DESCRIPTION
**Pull request by boydc2014**
Fix #2258

### CODE REVIEW

**Comment by benbrown** [comment]
Should this also clear the state variable used to track the message at the end of the process?

**Comment by boydc2014** [comment]
> Should this also clear the state variable used to track the message at the end of the process?

It's cleared\or reset, next time the preBubble is called

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs [hunk 1]
```diff
@@ -179,7 +185,17 @@ protected override async Task<bool> OnPreBubbleEvent(DialogContext dc, DialogEve
...
                     var state = await this.RecognizeInput(dc, true).ConfigureAwait(false);
-                    return state == InputState.Valid;
+                    var valid = state == InputState.Valid;
+                    if (!valid) 
+                    {
+                        // about to be interrupted
+                        dc.State.SetValue(INTERRUPTED_MESSAGE_ID, dc.Context.Activity.Id);
+                    } else
+                    {
+                        dc.State.SetValue(INTERRUPTED_MESSAGE_ID, null);
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs]: See here, @benbrown

**Comment by boydc2014** [comment]
> > Should this also clear the state variable used to track the message at the end of the process?
> 
> It's cleared\or reset, next time the preBubble is called

think a little more, i should clean it up incase sometime prebubble is not called.

**Comment by vishwacsena** [comment]
@boydc2014 - does not seem to work correctly. Source for the bot is in my branch. 

image

**Comment by boydc2014** [comment]
Hi @vishwac, it's working at you expected, you trained the number to have intent 
image
then in your interruption settings, there is no steps there, so it's working

image

**Comment by vishwacsena** [comment]
@boydc2014 its unclear what I need to change to have this scenarios work? 

```markdown
user: get started
bot: what is your age?
user: why do you ask?
bot: i need your age to …
bot: what is your age?
user: i’m not going to give you my age
bot: ok. I’ll assume 30. 
Bot: what is your name?
user: reset profile
bot: ok i’ve reset your profile
user: get started
bot: what is your age?
user: 35
bot: ok i have 35 as your age
bot: what is your name?
user: I’m not going to give you my name
bot: ok i’ll call you human. you can say reset profile to start over
bot: hello human, i have your age as 35.
```

**Comment by boydc2014** [comment]
Offline talked with @vishwac, this is not the behavior wanted, this is too strict, i will have a new PR to fix the issue to make sure it's at least working when step is null.

**Comment by boydc2014** [comment]
> @boydc2014 - does not seem to work correctly. Source for the bot is in my branch.
> 
> image

The reason it take age as name is becasue in your code, you specific age there
image

**Pull request closed without merging**